src/components/GuardianIdeal/GuardianIdealCarousel.tsx

```diff
@@ -0,0 +1,138 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import styles from './GuardianIdealCarousel.module.scss'
+import 'slick-carousel/slick/slick.css'
+import 'slick-carousel/slick/slick-theme.css'
+import GuardianIdealItem from './GuardianIdealItem'
+import { WidgetWithSettings } from 'components/Widget/Widget'
+import { IdealListItem, Widget } from 'types'
+import { useModalContext } from 'stores/modalContext'
+
+const CustomEllipse = ({ onClick }: any) => {
+  return (
+    <div className="news-carousel-container">
+      <button
+        onClick={onClick}
+        type="button"
+        title="slide"
+        className={styles.newsCarouselEllipse}
+      />
+    </div>
+  )
+}
+
+const GuardianIdealCarousel = ({
+  ideals,
+  widget,
+}: {
+  ideals: IdealListItem[]
+  widget: Widget
+}) => {
+  const { updateModalId, updateModalText, modalRef, updateWidget } =
+    useModalContext()
+
+  const sliderRef = useRef<Slider>(null)
+  const flags = useFlags()
+
+  const settings = {
+    dots: true,
+    fade: true,
+    accessibility: true,
+    adaptiveHeight: true,
+    infinite: true,
+    speed: 500,
+    slidesToShow: 1,
+    slidesToScroll: 1,
+    arrows: false,
+    dotsClass: `slick-dots ${styles.dots}`,
+    appendDots: (dots: React.ReactNode) => {
+      return (
+        <div>
+          <button
+            type="button"
+            className={styles.carouselArrow}
+            aria-label="previous guardian ideal"
+            data-testid="slick-prev"
+            onClick={() => {
+              if (sliderRef.current) {
+                return sliderRef.current.slickPrev()
+              }
+            }}>
+            <FontAwesomeIcon size="lg" icon="angle-left" />
+          </button>
+          <ul
+            style={{
+              margin: '0px',
+            }}>
+            {' '}
+            {dots}{' '}
+          </ul>
+          <button
+            type="button"
+            className={styles.carouselArrow}
+            aria-label="next guardian ideal"
+            data-testid="slick-next"
+            onClick={() => {
+              if (sliderRef.current) {
+                return sliderRef.current.slickNext()
+              }
+            }}>
+            <FontAwesomeIcon size="lg" icon="angle-right" />
+          </button>
+        </div>
+      )
+    },
+    customPaging: () => <CustomEllipse />,
+  }
+
+  const handleConfirmRemoveSection = () => {
+    updateModalId('removeSectionModal')
+    updateModalText({
+      headingText: 'Are you sure you’d like to delete this section?',
+      descriptionText:
+        'You can re-add it to your My Space from the Add Section menu.',
+    })
+
+    const widgetState: Widget = {
+      _id: widget._id,
+      title: widget.title,
+      type: 'GuardianIdeal',
+    }
+
+    updateWidget(widgetState)
+
+    modalRef?.current?.toggleModal(undefined, true)
+  }
+
+  return (
+    <>
+      {flags && flags.guardianIdealCarousel && (
```

Comment:
Suggestion: I don't think this is the right place to check the LD flag. If the flag is off I am still able to see `Add Guardian Ideal` in the add section menu. If I add the section I still get an empty div to hold the guardian ideal. Probably should move the check up so that the guardian ideal div doesn't show entirely and the option to add it also doesn't show if the flag is off.

![SCR-20230222-ixb](https://user-images.githubusercontent.com/940173/220765111-2f5a8b5b-7798-460d-ab35-897bce6e2005.png)

Reply:
+1 -- I was getting an empty placeholder div because I didn't have the LD flag set up properly locally. Makes sense to me that we shouldn't show the menu options if it's off.

Reply:
Good catch! May need to evaluate the flag in two places, since the button for adding it to My Space is in `AddWidget` I believe. If I need to do that I'll add a comment explaining it.